annlite/storage/table.py

```diff
@@ -336,17 +338,20 @@ def delete_by_offset(self, offset: int):
 
         :param offset: The offset of the doc
         """
-        sql = f'UPDATE {self.name} SET _deleted = 1 WHERE _id = ?'
+        # sql = f'UPDATE {self.name} SET _deleted = 1 WHERE _id = ?'
+        sql = f'DELETE FROM {self.name} WHERE _id = ?'
         self._conn.execute(sql, (offset + 1,))
         self._conn.commit()
 
     def undo_delete_by_offset(self, offset: int):
```

Comment:
If we use hard delete, there is no way to undo it. Thus, this function `undo_delete_by_offset` should be removed.